pyop2/compilation.py

```diff
@@ -285,65 +370,66 @@ def get_so(self, jitmodule, extension):
                     with open(cname, "w") as f:
                         f.write(jitmodule.code_to_compile)
                     # Compiler also links
-                    if self._ld is None:
-                        cc = [self._cc] + self._cppargs + \
-                             ['-o', tmpname, cname] + self._ldargs
+                    if not self.ld:
+                        cc = (compiler,) \
+                            + compiler_flags \
+                            + ('-o', tmpname, cname) \
+                            + self.ldflags
                         debug('Compilation command: %s', ' '.join(cc))
-                        with open(logfile, "w") as log:
-                            with open(errfile, "w") as err:
-                                log.write("Compilation command:\n")
-                                log.write(" ".join(cc))
-                                log.write("\n\n")
-                                try:
-                                    if configuration['no_fork_available']:
-                                        cc += ["2>", errfile, ">", logfile]
-                                        cmd = " ".join(cc)
-                                        status = os.system(cmd)
-                                        if status != 0:
-                                            raise subprocess.CalledProcessError(status, cmd)
-                                    else:
-                                        subprocess.check_call(cc, stderr=err,
-                                                              stdout=log)
-                                except subprocess.CalledProcessError as e:
-                                    raise CompilationError(
-                                        """Command "%s" return error status %d.
+                        with open(logfile, "w") as log, open(errfile, "w") as err:
+                            log.write("Compilation command:\n")
+                            log.write(" ".join(cc))
+                            log.write("\n\n")
+                            try:
+                                if configuration['no_fork_available']:
```

Comment:
This configuration option is another one of the meaningless ones so should probably get deleted.

Reply:
I was under the impression that on a system that prohibited forking, this would (at least in some cases) allow you to invoke the compiler.